tools/mage/general.go

```diff
@@ -33,3 +34,11 @@ func targetError(err error) error {
 	}
 	return nil
 }
+
+func pathExists(filename string) bool {
+	_, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
```

Comment:
This is technically incorrect. If you don't have permission to read the dir in which you're looking for the file, it would return that the file exists, while you don't actually know. 

Future developers who see that there is a `pathExists` in this package might use it incorrectly, so I don't think it's a good idea to do it this way.

Also, you can shorten L40-43 to `return !os.IsNotExist(err)`, which makes me wonder if it's really worth that one line you might save in the places where you'd use this.